kite_provider/provider.py

```diff
@@ -0,0 +1,229 @@
+# -*- coding: utf-8 -*-
+
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""Kite completion HTTP client."""
+
+# Standard library imports
+import logging
+import functools
+import os
+import os.path as osp
+
+# Qt imports
+from qtpy.QtCore import Slot
+from qtpy.QtWidgets import QMessageBox
+
+# Local imports
+from kite_provider.client import KiteClient
+from kite_provider.utils.status import (
+    check_if_kite_running, check_if_kite_installed)
+from kite_provider.widgets import (KiteStatusWidget)
+
+#Spyder imports
+from spyder.api.config.decorators import on_conf_change
+from spyder.config.base import _, running_under_pytest, get_module_data_path
+from spyder.plugins.completion.api import SpyderCompletionProvider
+from spyder.plugins.mainmenu.api import ApplicationMenus, ToolsMenuSections
+from spyder.utils.icon_manager import ima
+from spyder.utils.image_path_manager import IMAGE_PATH_MANAGER
+from spyder.utils.programs import run_program
+
+
+logger = logging.getLogger(__name__)
+
+
+class KiteProviderActions:
+    Installation = 'kite_installation'
+
+
```

Comment:
I think we can remove this since no action to install Kite should be available now:
```suggestion
```